File: TypeScript/design_parking_system/1603.ts
class ParkingSystem {
  big: number;
  medium: number;
  small: number;
  constructor(big: number, medium: number, small: number) {
    this.big = big;
    this.medium = medium;
    this.small = small;
  }

  addCar(carType: number): boolean {
    if (carType === 1) {
      return this.big-- > 0;
    } else if (carType === 2) {
      return this.medium-- > 0;
    } else {
      return this.small-- > 0;
    }
  }
}

export default ParkingSystem;

File: Golang/design_parking_system/1603.go
package design_parking_system

type ParkingSystem struct {
	small  int
	medium int
	big    int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem{small, medium, big}
}

func (p *ParkingSystem) AddCar(carType int) bool {
	if carType == 1 && p.big > 0 {
		p.big--
		return true
	}
	if carType == 2 && p.medium > 0 {
		p.medium--
		return true
	}
	if carType == 3 && p.small > 0 {
		p.small--
		return true
	}
	return false
}
